### ISSUE TITLE
[ADAP-1007] [Feature] Cast dataproc batch runtime_config properties to strings

### ISSUE THREAD

**Issue by jimmyshah** [dbt-labs/dbt-bigquery#1001]
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

While trying to configure Dataproc Serverless jobs via `profile.yml`, I keep getting an error unless I use quotes around each key-value pair, which doesn't match [dbt's documentation](https://docs.getdbt.com/docs/core/connect-data-platform/bigquery-setup#running-python-models-on-dataproc).

What works for me:

```yml
runtime_config:
          properties:
           'spark.executor.instances': '4'
            'spark.driver.memory': '4g'
```

### Expected Behavior

According to [dbt's documentation](https://docs.getdbt.com/docs/core/connect-data-platform/bigquery-setup#running-python-models-on-dataproc), Dataproc Serverless jobs can be configured in the dbt `profiles.yml`  file, without quotes. In the example provided on the site, `runtime_config` `properties` can be specified in yaml as such:

```yml
runtime_config:
          properties:
            spark.executor.instances: 3
            spark.driver.memory: 1
```

### Steps To Reproduce

1.  You have dbt python models already in your project
2.  Add runtime_config properties for serverless jobs without quotes
3.  Run your dbt-py model

### Relevant log output

```shell
Unable to parse dataproc_batch as valid batch specification. See https://cloud.google.com/dataproc-serverless/docs/reference/rpc/google.cloud.dataproc.v1#google.cloud.dataproc.v1.Batch. Failed to parse runtime_config field: Failed to parse properties field: expected string or bytes-like object..
```

### Environment

```markdown
- OS: Mac OS Sonoma 14.1
- Python: 3.9.6
- dbt-core: 1.6.2
- dbt-bigquery: 1.6.4
```

**Comment by dbeatty10** [comment]
Thanks for reporting this @jimmyshah !

Did only single quotes work for you, or did double quotes work also, like this?
```yaml
        runtime_config:
          properties:
            "spark.executor.instances": "4"
            "spark.driver.memory": "4g"
```

Does this work for you?
```yaml
        runtime_config:
          properties:
            spark.executor.instances: '4'
            spark.driver.memory: '4g'
```

Or did the `spark.executor.instances` and `spark.driver.memory` keys need to be quoted as well?

Ultimately, I'm curious if maybe only integers need to be quoted, but everything else doesn't need to be quoted?

**Comment by jimmyshah** [comment]
Hi @dbeatty10!

- Example 1: Double quotes also works
- Example 2: Only quoting the values also works

I also tried having only the '4' in quotes, and that also worked. This makes me think your theory is correct, that only integers need to be quoted.

**Comment by dbeatty10** [comment]
Thanks for checking all those @jimmyshah !

I'm going to relabel this as a feature since technically dbt-biquery is working properly but isn't user-friendly. The docs are erroneous though.

To address both short-term and long-term, we should do the following:
1. Update the documentation to show quotes around integers within `properties` of `runtime_config`
2. Cast to a string within dbt-bigquery regardless if the user quotes or not

### 1. Update the documentation

I opened a PR to update the documentation:
- https://github.com/dbt-labs/docs.getdbt.com/pull/4406

### 2. Cast to a string within dbt-bigquery

Replace the [`passthrough`](https://github.com/Fatal1ty/mashumaro/blob/b75801b8eb92a85a56007169097f13cea9e17f86/mashumaro/helper.py#L47-L58)  serialization [here](https://github.com/dbt-labs/dbt-bigquery/blob/f6e26d0073fbf3830457532f57b3b59e60af0535/dbt/adapters/bigquery/connections.py#L153-L158) with a custom serialization [[1](https://github.com/Fatal1ty/mashumaro#serializationstrategy), [2](https://github.com/Fatal1ty/mashumaro#serialization_strategy-config-option)] that serializes `properties` values to `str`.

Alternatively, casting values of `properties` to string can be done [here](https://github.com/dbt-labs/dbt-bigquery/blob/f6e26d0073fbf3830457532f57b3b59e60af0535/dbt/adapters/bigquery/python_submissions.py#L177-L179).

**Comment by jimmyshah** [comment]
Thank you for the quick and thorough response @dbeatty10!

It looks like your docs PR has been merged, so the short-term fix should be in!

That said, I'm happy to raise a PR to address the underlying issue. Thank you for providing some guidance on how to best implement the change. I am leaning towards the second solution you proposed of casting the `properties` to strings.

**Comment by dbeatty10** [comment]
Awesome @jimmyshah would be great if you want to raise a PR for this.

As a heads-up, the test case(s) always seem to be the most time-consuming part in the code review process.

Other things to be aware of:
- Add `resolves #1001` to the top of the PR description to link it to this issue
- Install `changie` and run `changie new` as described [here](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md#adding-changelog-entry) in order to create a changelog entry

**Comment by jimmyshah** [comment]
Hi @dbeatty10! Thank you for the heads up, definitely helpful!

I opened PR #1008  that (I think?) resolves the issue! For `changie`, I created a feature entry since we changed this issue from a bug to a feature.  If there is anything else I can do from my end, let me know!
